frontend/public/components/utils/tile-view-page.jsx

```diff
@@ -495,12 +494,46 @@ export class TileViewPage extends React.Component {
   }
 
   componentDidMount() {
-    const { items, filterGroups, getAvailableFilters, groupByTypes } = this.props;
+    const {
+      items,
+      filterGroups,
+      getAvailableFilters,
+      groupByTypes,
+      storeFilterKey,
+      filterRetentionPreference,
+      defaultActiveFilters,
+    } = this.props;
     const { categories } = this.state;
     const availableFilters = getAvailableFilters(defaultFilters, items, filterGroups);
-
+    if (!(storeFilterKey && localStorage.getItem(storeFilterKey)) && defaultActiveFilters) {
+      _.each(Object.keys(defaultActiveFilters), (filterGroup) => {
+        _.each(defaultActiveFilters[filterGroup], (key) => {
+          availableFilters[filterGroup][key].active = true;
+        });
+      });
+    }
     const activeValues = getActiveValuesFromURL(availableFilters, filterGroups, groupByTypes);
 
+    if (storeFilterKey && localStorage.getItem(storeFilterKey)) {
+      try {
+        const lastFilters = JSON.parse(localStorage.getItem(storeFilterKey));
+        if (lastFilters && activeValues.activeFilters) {
+          filterRetentionPreference &&
```

Comment:
Use `if` statements instead of `&&` conditions to make code intent much more clear.

Reply:
updated